File: library-app/src/main/java/com/example/libraryapp/controller/book/BookController.java
package com.example.libraryapp.controller.book;

import com.example.libraryapp.dto.book.request.BookLoanRequest;
import com.example.libraryapp.dto.book.request.BookRequest;
import com.example.libraryapp.dto.book.request.BookReturnRequest;
import com.example.libraryapp.service.book.BookService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class BookController {

    private final BookService bookService;

    public BookController(BookService bookService) {
        this.bookService = bookService;
    }

    @PostMapping("/book")
    public void saveBook(@RequestBody BookRequest request) {
        bookService.saveBook(request);
    }

    @PostMapping("/book/loan")
    public void loanBook(@RequestBody BookLoanRequest request) {
        bookService.loanBook(request);
    }

    @PutMapping("/book/return")
    public void returnBook(@RequestBody BookReturnRequest request) {
        bookService.returnBook(request);
    }
}

File: library-app/src/main/kotlin/com/example/libraryapp/controller/book/BookController.kt
package com.example.libraryapp.controller.book

import com.example.libraryapp.dto.book.request.BookLoanRequest
import com.example.libraryapp.dto.book.request.BookRequest
import com.example.libraryapp.dto.book.request.BookReturnRequest
import com.example.libraryapp.service.book.BookService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class BookController(
    private val bookService: BookService
) {

    @PostMapping("/book")
    fun saveBook(@RequestBody request: BookRequest) {
        bookService.saveBook(request)
    }

    @PostMapping("/book/loan")
    fun loanBook(@RequestBody request: BookLoanRequest) {
        bookService.loanBook(request)
    }

    @PutMapping("/book/return")
    fun returnBook(@RequestBody request: BookReturnRequest) {
        bookService.returnBook(request)
    }
}